percy/percy.test.js

```diff
@@ -0,0 +1,247 @@
+jest.setTimeout(600000)
+
+// How long to wait for each page to fully render before taking a screenshot
+const PERCY_EXTRA_WAIT = 5000
+const percySnapshotWithWait = async (page, name, enableJavaScript) => {
+  await page.waitForTimeout(PERCY_EXTRA_WAIT)
+
+  const namePrefix = process.env.PERCY_OTP_CONFIG_OVERRIDE || 'Mock OTP1 Server'
+  await percySnapshot(page, `${namePrefix} - ${name}`, { enableJavaScript })
+}
+
+let browser
+const serveAbortController = new AbortController()
+const harAbortController = new AbortController()
+
+/**
+ * Loads a path
+ */
+async function loadPath(otpPath) {
+  const page = await browser.newPage()
+  const filePath = `http://localhost:${MOCK_SERVER_PORT}/#${otpPath}`
+  await Promise.all([
+    page.goto(filePath),
+    page.waitForNavigation({ waitUntil: 'networkidle2' })
+  ])
+
+  return page
+}
+
+beforeAll(async () => {
+  try {
+    // Build OTP-RR main.js using new config file
+    await execa('env', [
+      `YAML_CONFIG=${
+        process.env.PERCY_OTP_CONFIG_OVERRIDE || OTP_RR_TEST_CONFIG_PATH
+      }`,
+      `JS_CONFIG=${OTP_RR_TEST_JS_CONFIG_PATH}`,
+      'yarn',
+      'build'
+    ])
+
+    // grab ATL har file to tmp
+    if (process.env.HAR_URL) {
+      await execa('curl', [process.env.HAR_URL, '-s', '--output', 'mock.har'])
+    }
+  } catch (error) {
+    console.log(error)
+  }
+  console.log('Built OTP-RR and downloaded HAR data')
+
+  try {
+    execa('yarn', ['percy-serve', 'dist', '-p', MOCK_SERVER_PORT], {
+      signal: serveAbortController.signal
+    }).stdout.pipe(process.stdout)
+
+    // Launch mock OTP server
+    if (process.env.HAR_URL) {
+      execa('yarn', ['percy-har-express', '-p', '9999', 'mock.har'], {
+        signal: harAbortController.signal
+      }).stdout.pipe(process.stdout)
+    }
+
+    // Web security is disabled to allow requests to the mock OTP server
+    browser = await puppeteer.launch({
+      args: ['--disable-web-security']
+      // headless: false
+    })
+
+    // Fix time
+    browser.on('targetchanged', async (target) => {
+      const targetPage = await target.page()
+      const client = await targetPage.target().createCDPSession()
+      await client.send('Runtime.evaluate', {
+        expression:
+          'Date.now = function() { return 1646835742000; };Date.getTime = function() { return 1646835742000; }'
+      })
+    })
+  } catch (error) {
+    console.log(error)
+  }
+
+  // Give servers time to start up
+  await execa('sleep', ['5'])
+})
+
+afterAll(async () => {
+  try {
+    serveAbortController.abort()
+    harAbortController.abort()
+    await browser.close()
+  } catch (error) {
+    console.log(error)
+  }
+  console.log('Closed mock server and headless browser')
+})
+
+// Puppeteer can take a long time to load, espeically in some ci environments
+jest.setTimeout(600000)
+
+/* These fixed routes allow us to test features that the static html screenshots
+ * don't allow us to test. Unfortuantley, they don't currently work
```

Comment:
Typo "unfortunately".

Reply:
Replace this whole comment with a TODO comment to make percy work with transitive.js.